pkg/cloudprovider/providers/clc/clc.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clc
+
+import (
+	"fmt"
+	"io"
+
+	"encoding/base64"
+
+	"github.com/golang/glog"
+	"gopkg.in/gcfg.v1"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/types"
+)
+
+const (
+	// ProviderName clc for CenturyLinkCloud
+	ProviderName = "clc"
+)
+
+// CLCCloud is an implementation of Interface, LoadBalancer and Zones for CenturyLinkCloud.
+type CLCCloud struct {
+	clcClient CenturyLinkClient
+	clcLB     *clcProviderLB // cloudprovider's LoadBalancer interface is implemented here
+	clcConfig Config         // loaded in readConfig
+}
+
+func init() {
+	glog.Info("registering CLC provider")
+
+	cloudprovider.RegisterCloudProvider(ProviderName, func(config io.Reader) (cloudprovider.Interface, error) {
+		cfg, err := readConfig(config)
+		if err != nil {
+			glog.Info(fmt.Sprintf("CLC provider could not read config: err=%s", err))
+			return nil, err
+		}
+
+		newClient := makeCenturyLinkClient()
+
+		newCloud := CLCCloud{
+			clcClient: newClient,
+			clcLB:     makeProviderLB(newClient),
+			clcConfig: cfg,
+		}
+
+		newClient.GetCreds().CredsLogin(cfg.Global.Username, cfg.Global.Password) // try the login, but accept that it may fail
+		if !newCloud.clcLB.clcClient.GetCreds().IsValid() {
+			glog.Info("CLC: initial login is not valid, either it failed or was aliased")
```

Comment:
Two reasons:  the failure may be transient, and we don't want to prevent operation of an existing cluster just because we're having an issue that prevents creation of a load balancer.